app/controllers/projects_controller.rb

```diff
@@ -101,8 +100,24 @@ def set_properties
 
   def set_aggregation
     a = Event::AGGREGATIONS.key?(params[:agg]) ? Event::AGGREGATIONS[params[:agg]] : "day"
-    @day_not_allowed = time_period > 6
-    a = "week" if a == "day" && @day_not_allowed
+    @possible_aggregation = case params[:date]
+                            when "30d"
+                              ["d", "w", "m"] 
+                            when "this_m"
+                              ["d", "w", "m"] 
+                            when "6m"
+                              ["w", "m", "y"] 
+                            when "12m"
+                              ["w", "m", "y"] 
+                            when "7d"
+                              ["d", "w"]
+                            else
+                              ["d", "w", "m", "y"]
+                            end
+    a = "week" if a == "day" && !@possible_aggregation.include?("d")
+    a = "week" if a == "month" && !@possible_aggregation.include?("m")
+    a = "week" if a == "year" && !@possible_aggregation.include?("m") && !@possible_aggregation.include?("y")
+    a = "month" if a == "year" && !@possible_aggregation.include?("y")
```

Comment:
You can probably solve this in a more elegant way with a recursive function (a function that calls itself until it has found a solution), maybe somthing like this pseudo code:

```ruby
def find_next_possible_agg
  # return when it has found the desired valid aggregation
  return if valid_aggregation_found

  # do some stuff here
  
  # call itself for the next iteration of the recursive function if it hasn't found the solution yet
  find_next_possible_agg
end
```

Instead of using the string literals (e.g., `week`), it's better to use the constant `Event::AGGREGATIONS` like this `Event::AGGREGATIONS["w"]`. This way, if we change the value of `week` in the future, we only need to change it in one place.

Code style: Better to call `@possible_aggregations`(plural) since it usually contains more than one value.

Code style: Better to use `exlude?` than `! include?`

Reply:
solved it in a completely different way.
used @possible_aggregations instead of @possible_aggregation
and exclude? instead of !...include?